code/obj/structure.dm

```diff
@@ -299,19 +301,16 @@ obj/structure/ex_act(severity)
 		var/FloorIntact = T.intact
 		var/FloorBurnt = T.burnt
 		var/FloorName = T.name
-		var/oldmat = src.material
 
 		var/target_type = S.reinforcement ? /turf/simulated/wall/false_wall/reinforced : /turf/simulated/wall/false_wall
 
 		T.ReplaceWith(target_type, FALSE, FALSE, FALSE)
 		var/atom/A = src.loc
-		if(oldmat)
-			A.setMaterial(oldmat)
-		else
-			var/datum/material/M = getMaterial("steel")
-			A.setMaterial(M)
-
+		var/datum/material/M = getMaterial("steel")
 		var/turf/simulated/wall/false_wall/FW = A
+
+		FW.setMaterial(S.material ? S.material : M, copy = FALSE)
```

Comment:
oh yeah that, I think that was just from me moving code from what it was before into ? : , It should be S.material or M, copy = FALSE so that it doesn't run the copymaterial proc im pretty sure, but I'm not actually sure the priority of ? : now that I think about it. does the M, copy = FALSE need to be inside parenthesis
i.e. FW.setMaterial(S.material ? S.material : (M, copy = FALSE))